Cherry Pick for Mozilla Bug:
	436133 - "Cookies build failure on hppa"
	https://bugzilla.mozilla.org/show_bug.cgi?id=436133

From Attachment:
	https://bugzilla.mozilla.org/attachment.cgi?id=322801

--
Index: configure.in
===================================================================
RCS file: /cvsroot/mozilla/configure.in,v
retrieving revision 1.1992
diff -u -b -B -u -8 -p -r1.1992 configure.in
--- a/configure.in	5 May 2008 21:08:14 -0000	1.1992
+++ b/configure.in	28 May 2008 16:45:58 -0000
@@ -1237,16 +1237,20 @@ case "$OS_TEST" in
 powerpc* | ppc)
     CPU_ARCH=ppc
     ;;
 
 Alpha | alpha | ALPHA)
     CPU_ARCH=Alpha
     ;;
 
+hppa* | parisc)
+    CPU_ARCH=hppa
+    ;;
+
 sun4u | sparc*)
     CPU_ARCH=sparc
     ;;
 
 x86_64 | ia64)
     CPU_ARCH="$OS_TEST"
     ;;
 esac
@@ -1287,19 +1291,19 @@ if test "$GNU_CC"; then
     # -Wall - turn on all warnings
     # -pedantic - make compiler warn about non-ANSI stuff, and
     #             be a little bit stricter
     # Warnings slamm took out for now (these were giving more noise than help):
     # -Wbad-function-cast - warns when casting a function to a new return type
     # -Wconversion - complained when char's or short's were used a function args
     # -Wshadow - removed because it generates more noise than help --pete
     _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Wall -W -Wno-unused -Wpointer-arith"
-    if test "$CPU_ARCH" != "ia64" && test "$CPU_ARCH" != "sparc" \
+    if test "$CPU_ARCH" != "hppa" && test "$CPU_ARCH" != "ia64" && test "$CPU_ARCH" != "sparc" \
        && test -z "$INTEL_CC"; then
-        # don't use -Wcast-align on ia64 or sparc, it's noisy on those platforms
+        # don't use -Wcast-align on hppa, ia64 or sparc, it's noisy on those platforms
         # icc doesn't support this flag.
         _WARNINGS_CFLAGS="${_WARNINGS_CFLAGS} -Wcast-align"
     fi
 
     dnl Turn pedantic on but disable the warnings for long long
     _PEDANTIC=1
 
     if test -z "$INTEL_CC"; then
@@ -1327,19 +1331,19 @@ else
     _DEFINES_CFLAGS='$(ACDEFINES) -D_MOZILLA_CONFIG_H_ -DMOZILLA_CLIENT'
 fi
 
 if test "$GNU_CXX"; then
     # FIXME: Let us build with strict aliasing. bug 414641.
     CXXFLAGS="$CXXFLAGS -fno-strict-aliasing"
     # Turn on GNU specific features
     _WARNINGS_CXXFLAGS="${_WARNINGS_CXXFLAGS} -Wall -Wconversion -Wpointer-arith -Woverloaded-virtual -Wsynth -Wno-ctor-dtor-privacy -Wno-non-virtual-dtor"
-    if test "$CPU_ARCH" != "ia64" && test "$CPU_ARCH" != "sparc" \
+    if test "$CPU_ARCH" != "hppa" && test "$CPU_ARCH" != "ia64" && test "$CPU_ARCH" != "sparc" \
        && test -z "$INTEL_CC"; then
-        # don't use -Wcast-align on ia64 or sparc, it's noisy on those platforms
+        # don't use -Wcast-align on hppa, ia64 or sparc, it's noisy on those platforms
         # icc doesn't support this flag.
         _WARNINGS_CXXFLAGS="${_WARNINGS_CXXFLAGS} -Wcast-align"
     fi
 
     _DEFINES_CXXFLAGS='-DMOZILLA_CLIENT -include $(DEPTH)/mozilla-config.h'
     _USE_CPP_INCLUDE_FLAG=1
 else
     _DEFINES_CXXFLAGS='-DMOZILLA_CLIENT -D_MOZILLA_CONFIG_H_ $(ACDEFINES)'
